### ISSUE TITLE
Build a fintech app - tweaks

### ISSUE THREAD

**Issue by Cupofassam** [rehive/dashboard-angular#35]
Hey Aquib,
We need a few tweaks to the "Build a fintech app" section of the dashboard.

1. The Next button - when a user is not able to click it, please make it this shade of blue: 
![image](https://user-images.githubusercontent.com/22153546/36731060-312a30f6-1bd2-11e8-866d-f5354724112a.png) You can keep the "do not enter" icon which the cursor turns into. As soon as it becomes clickable, please make it turn dark blue like this: 
![image](https://user-images.githubusercontent.com/22153546/36731085-44fce2a4-1bd2-11e8-809b-68e905100a0d.png)

2. When a user adds a group, display whether a group is public or default under the group Label and name. The Label is on the left and the name on the right.
![image](https://user-images.githubusercontent.com/22153546/36734172-5fed7188-1bdb-11e8-8589-63c6f148748b.png)
![image](https://user-images.githubusercontent.com/22153546/36736591-b251516e-1be1-11e8-9bf3-9a217c07a5b4.png)

(Obviously service and admin won't have these, just as example)

3. Update the copy on the Groups page. Please change bullet point 4 to:

`Make a group Default if you want new users to be automatically added upon user creation. Only one group can be set as the Default.`

4. On Currencies page, when a user has not added any currencies, the copy currently reads "Currencies selected:  ". Please change this to say `No currencies selected yet.` instead. It can then change to "Currencies selected: USD, EUR" etc once a user has begun selecting currencies.

5. On the Currencies page, please ensure that there is no comma after the last currency selected. So if a user has selected USD and XBT the copy under the currencies selection box should read `Currencies selected: USD, XBT` rather than `Currencies selected: USD, XBT,` 
![image](https://user-images.githubusercontent.com/22153546/36775930-b0363206-1c6c-11e8-9607-765095993174.png)

6. Update the copy on the Currencies page. Please make an additional bullet point and make it the first line of copy:
`Add which currencies your business can transact in.`

7. For all of the onboarding hint copy on the left, please increase the line height.

8. Adjust the copy of last bullet point of Transaction subtypes to:

`Subtypes can be used to define business logic for marked transactions in the dashboard. E.g. A debit subtype named “withdrawal” can apply fees to only debit transactions marked with subtype withdrawal. `

9. Wherever a user is entering a name and a label, please automatically generate the label from the name as the user is typing the name, kind of like when CMS's auto generate alt tags from file names.

**Comment by Cupofassam** [comment]
Updated my issue with a few more tweaks

**Comment by Cupofassam** [comment]
Thanks Aquib :) some feedback:

1. Please hide any currency from the selection box which a user has already added.
2. In groups, remove the display of the label, just have the name there without brackets.
3. On the left of the X in the "added groups" section, please add a small edit pencil icon. Clicking on this will allow the item to be edited rather than clicking.

**Comment by Cupofassam** [comment]
Okay, I have looked through the onboarding flow and I think there's too much wall-of-texting and upfront info. I want to move the text into informational popups users can look at if they want. Changing Groups, Currencies and Account configuration pages.

### All

1. A weird change to apply to all pages. The "x" that let's a user remove a group/currency/account/subtype they have added is too long compared to its width. Can we make a more square X?

### Groups 

![image](https://user-images.githubusercontent.com/22153546/38688584-bd716d12-3e79-11e8-8aa0-a4f3c996ec1b.png)

1. There will be a small "i" icon next to Public, Default and Rehive system groups. Hovering over this "i" will display an informational popup.
2. `service` and `admin` groups now move to a separate Rehive system groups box to get them out of the way.
3. Here is the copy for the popups:

**Public:** 
`Make a group Public if you would like the group to be available as a choice in the registration API endpoint.`

**Default:** 
`Make a group Default if you want new users to be automatically added to it when they register. Only one group can be set as the Default.`

**Rehive system groups:** 
`admin group is automatically added and the company owner placed inside. service group is required to add utility services on Rehive.`

**note**: please make **_Public, Default, admin_** and **_service_** all bold + italic.

4. Here is the updated copy that should be on the side in bullet points:

- `Create groups to categorize and configure business rules for your users.`
- `You can configure tiers, tier limits, tier requirements, fees, and admin permissions per group later on.`

### Currencies

1. Only one change here - get rid of this copy please:

- `Choose currencies from a predefined list which includes currency codes, symbols, names, and divisibility. Divisibility refers to the decimal places for a currency.`

### Account configurations

![image](https://user-images.githubusercontent.com/22153546/38689012-c026ae54-3e7a-11e8-8ba8-dded8bf26d49.png)

1. Please add "i" icons next to Primary and Default.
2. Informational copy:

**Primary:** (should be bullet points)
` All money transferred to users will go to their Primary account.`
`There can only be one Primary account per group.`
`You can't make transfers to a user if they don't have a Primary account.`

**Default:**
`Making an account Default means all users in a selected group will automatically have this account and its configurations when they register.`

**note:** please make **_Primary_** and **_Default_** all bold + italic.

3. New copy on the left:

- `Set up templates for account configurations per user group. These are applied to users in that group when they resister.`
- `You can add users to groups you create at any time.`
- `Select which currencies the account can transact in.`

4. I don't like the way the added accounts looks right now with the name and label. Please make it more like Groups where we show the account name and then whether it is Default or Primary underneath.

**Comment by Cupofassam** [comment]
Feedback:

### Groups

1. When I hover over the Rehive system groups i icon, the whole tooltip is in italics instead of just "admin" and "service".
2. Please left align all text in the tips
3. Please make the boxes for the tips a bit longer - maybe 10% more (each tooltip currently has 1 hanging word at the end, I want to get that word onto the previous line in all cases)

### Account configurations

1. Left align tooltip copy
2. The "business" label is in a bigger font compare to the "defaulto" label on the groups page, please make them the same:

![image](https://user-images.githubusercontent.com/22153546/38736050-092d4940-3f2b-11e8-8185-5b364f265cac.png)

![image](https://user-images.githubusercontent.com/22153546/38736060-134c4458-3f2b-11e8-86ac-f7337e1f6e29.png)

**Comment by Cupofassam** [comment]
Feedback:

### Accounts

1. Please slightly indent all the added accounts under a group (you can move edit button slightly closer to the remove X)
2. If an account name is really long it pushes down the edit button, please make it wrap before that so the edit button doesn't move

![image](https://user-images.githubusercontent.com/22153546/38799899-0681d7b4-4166-11e8-9034-b67749bede3a.png)

3. Please make Primary's tooltip box the same length as Default's(or until the last word in the first bullet point is on  the same line.
4. Remove extra spacing at the bottom of Primary's tooltip

![image](https://user-images.githubusercontent.com/22153546/38799991-4dce492c-4166-11e8-8b64-a8f35d5c7a05.png)
![image](https://user-images.githubusercontent.com/22153546/38800018-690b8826-4166-11e8-804c-e296cdc9a9ed.png)

**Comment by Cupofassam** [comment]
I have 1 possible bug report, 2 feedback and 1 question.

**bug:**

I switched companies and tried to add USD and ZAR, but they were hidden from the currency list. It seems to be saving my previous company's selection and carrying it over to the new company I logged in with. When I hard refreshed it was ok, but would one have to hard refresh each time?

**feedback:**

1. If I type my account configuration name as "business_account", the label becomes "Business_account". Can we make the label generator ignore "-" and "_"? We can allow a user to enter those characters if they want to amend the label, but it should ignore it by default when it's in the name.

2. The Primary tooltip box can be even wider, until "account" is on the same line as "Primary" in the first bullet point.

**question:**

How long would it take to add a search bar to the currencies? I don't think we should do it now but it would be nice in the future I think.
